core/rawdb/freezer_table_test.go

```diff
@@ -579,7 +579,7 @@ func TestOffset(t *testing.T) {
 		// It should be fine to fetch 4,5,6
 		if got, err := f.Retrieve(4); err != nil {
 			t.Fatal(err)
-		} else if exp := getChunk(20, 0xbb); !bytes.Equal(got, exp) {
+		} else if exp := getChunk(16, 0xbb); !bytes.Equal(got, exp) {
```

Comment:
|-----------------|
| fileno | offset |
|--------+--------|
|  002   |  004   | 
|  002   |  020   | 

should be `Chunk(16, 0xbb)`

Reply:
Why should it be? We wrote `		f.Append(4, getChunk(20, 0xbb))` earlier, why should we expect `16` bytes when reading it back? 
The first offset as shown in the table is not an _actual_ offset. We know that the first offset is always zero, so we use that field for storing the item-number of the first item.

Reply:
https://github.com/ethereum/go-ethereum/blob/14589332407a1e75afc6447fd5ab74cc3b0672d3/core/rawdb/freezer_table_test.go#L552-L557

the test overwrites the first offset here, so a truncated item returned.
seems the original design is intended to do so.